client/client_test.go

```diff
@@ -99,22 +93,59 @@ func Test_GetFlyteHealthCheckURL_ShouldReturnErrorWhenItCannotGetHealthCheckURLF
 }
 
 func Test_TakeAction_ShouldReturnSpecificErrorTypeAndMessageWhenResourceIsNotFound(t *testing.T) {
-	handler := func(w http.ResponseWriter, r *http.Request) { w.WriteHeader(http.StatusNotFound) }
-	server := httptest.NewServer(http.HandlerFunc(handler))
-	defer server.Close()
+	ts := mockServer(http.StatusNotFound, "")
+	defer ts.Close()
 
-	u, _ := url.Parse(server.URL + "/take/action/url")
+	c := newTestClient(ts.URL, t)
+	u, err := url.Parse(ts.URL + "/take/action/url")
+	require.NoError(t, err)
 
-	c := &client{takeActionURL: u, httpClient: &http.Client{
-		Timeout: 5 * time.Second,
-	}}
+	c.takeActionURL = u
+	_, err = c.TakeAction()
 
-	_, e := c.TakeAction()
+	require.IsType(t, NotFoundError{}, err)
+	assert.EqualError(t, err, fmt.Sprintf("Resource not found at %s/take/action/url", ts.URL))
+}
 
-	if assert.Error(t, e) {
-		assert.IsType(t, NotFoundError{}, e)
-		assert.Equal(t, fmt.Sprintf("Resource not found at %s/take/action/url", server.URL), e.Error())
-	}
+//should register pack with API, and have populated links
```

Comment:
No need for this comment...but you could change the test name to 'Test_CreatePack_ShouldRegisterPackWithApiAndPopulateClientWithLinks' or similar

Reply:
done